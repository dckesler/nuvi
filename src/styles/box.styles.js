export const card = {
	borderRadius: "4px",
	boxShadow: "2px 2px 1px rgba(0, 0, 0, 0.2)",
	padding: "16px",
}
export const nuvi = {
	background: "#54C7DD",
}
export const twitter = {
	background: "#1DA1F3",
};
export const instagram = {
	background: "linear-gradient(145deg, #3051F1 10%, #C92BB7 70%) no-repeat",
};
export const facebook = {
	background: "#3B5998",
};
export const tumblr = {
	background: "#37475C",
};
