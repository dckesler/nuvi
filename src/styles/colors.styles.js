export const blue = "#54C7DD";
export const green = '#83C05A';
export const red = "#ED7B90";
export const inactive = "#646464";
export const twitter = "#1DA1F3";
export const facebook = "#3B5998";
export const instagram = "#8F39CC";
export const tumblr = "#37475C";
