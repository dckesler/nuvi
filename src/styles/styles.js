import * as boxes from './box.styles.js';
import * as colors from './colors.styles.js';
import * as texts from './text.styles.js';
import { css, StyleSheet } from 'aphrodite';

export default {
	texts,
	boxes,
	colors,
	toClass(style) {
		return css(StyleSheet.create(style));
	}
}
