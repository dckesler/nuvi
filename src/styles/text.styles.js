import * as colors from './colors.styles.js';

export const header = {
	fontSize: "32px",
	fontWeight: "500",
}

export const subheader = {
	fontSize: "24px",
	fontWeight: "400",
}

export const body = {
	fontSize: "16px",
	fontWeight: "300",
}

export const caption = {
	fontSize: "12px",
	fontWeight: "200",
}

export const link = {
	textDecoration: "none",
	cursor: "pointer",
	color: colors.blue,
	padding: "4px",
	borderRadius: "4px",
	":hover": {
		background: "rgba(0, 0, 0, .1)"
	}
}
