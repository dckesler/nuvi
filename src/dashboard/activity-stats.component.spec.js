import React from 'react';
import { shallow } from 'enzyme';
import { css } from 'aphrodite';
import Dashboard, { styles } from './dashboard.component.js';
import ActivityStats from './activity-stats.component.js';
import { colors } from 'styles';

describe('activity-stats.component', () => {
	it('should render properly', () => {
		const wrapper = shallow(
			<ActivityStats likes={1} shares={2} comments={3} id={4} />
		);
		expect(wrapper).toMatchSnapshot();
	});
	it('should adjust icon colors based on interaction', () => {
		const wrapper = shallow(
			<ActivityStats likes={1} shares={2} comments={3} id={4} />
		);
		expect(wrapper.find(`.icon-like`).props().style.color).toBe(colors.inactive);
		expect(wrapper.find(`.icon-share`).props().style.color).toBe(colors.inactive);
		expect(wrapper.find(`.icon-comment`).props().style.color).toBe(colors.inactive);
		wrapper.setState({like: true, share: true, comment: true});
		expect(wrapper.find(`.icon-like`).props().style.color).toBe(colors.green);
		expect(wrapper.find(`.icon-share`).props().style.color).toBe(colors.green);
		expect(wrapper.find(`.icon-comment`).props().style.color).toBe(colors.green);
	});
})
