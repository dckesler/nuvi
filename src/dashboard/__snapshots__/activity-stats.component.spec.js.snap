exports[`activity-stats.component should render properly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="activityStats_13efbjt">
    <div
        className="stat_y6hf3w">
        1
        <div
            className="statIcon_1id1dbo"
            onClick={[Function]}>
            <i
                className="icon-like"
                style={
                    Object {
                        "color": "#646464",
                      }
                } />
        </div>
    </div>
    <div
        className="stat_y6hf3w">
        2
        <div
            className="statIcon_1id1dbo"
            onClick={[Function]}>
            <i
                className="icon-share"
                style={
                    Object {
                        "color": "#646464",
                      }
                } />
        </div>
    </div>
    <div
        className="stat_y6hf3w">
        3
        <div
            className="statIcon_1id1dbo"
            onClick={[Function]}>
            <i
                className="icon-comment"
                style={
                    Object {
                        "color": "#646464",
                      }
                } />
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className="activityStats_13efbjt">
      <div
            className="stat_y6hf3w">
            1
            <div
                  className="statIcon_1id1dbo"
                  onClick={[Function]}>
                  <i
                        className="icon-like"
                        style={
                              Object {
                                    "color": "#646464",
                                  }
                        } />
            </div>
      </div>
      <div
            className="stat_y6hf3w">
            2
            <div
                  className="statIcon_1id1dbo"
                  onClick={[Function]}>
                  <i
                        className="icon-share"
                        style={
                              Object {
                                    "color": "#646464",
                                  }
                        } />
            </div>
      </div>
      <div
            className="stat_y6hf3w">
            3
            <div
                  className="statIcon_1id1dbo"
                  onClick={[Function]}>
                  <i
                        className="icon-comment"
                        style={
                              Object {
                                    "color": "#646464",
                                  }
                        } />
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ActivityStats
        comments={3}
        id={4}
        likes={1}
        shares={2} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ActivityStats {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "comments": 3,
          "id": 4,
          "likes": 1,
          "shares": 2,
        },
        "refs": Object {},
        "state": Object {},
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="activityStats_13efbjt">
          <div
                    className="stat_y6hf3w">
                    1
                    <div
                              className="statIcon_1id1dbo"
                              onClick={[Function]}>
                              <i
                                        className="icon-like"
                                        style={
                                                  Object {
                                                            "color": "#646464",
                                                          }
                                        } />
                    </div>
          </div>
          <div
                    className="stat_y6hf3w">
                    2
                    <div
                              className="statIcon_1id1dbo"
                              onClick={[Function]}>
                              <i
                                        className="icon-share"
                                        style={
                                                  Object {
                                                            "color": "#646464",
                                                          }
                                        } />
                    </div>
          </div>
          <div
                    className="stat_y6hf3w">
                    3
                    <div
                              className="statIcon_1id1dbo"
                              onClick={[Function]}>
                              <i
                                        className="icon-comment"
                                        style={
                                                  Object {
                                                            "color": "#646464",
                                                          }
                                        } />
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="activityStats_13efbjt">
          <div
                    className="stat_y6hf3w">
                    1
                    <div
                              className="statIcon_1id1dbo"
                              onClick={[Function]}>
                              <i
                                        className="icon-like"
                                        style={
                                                  Object {
                                                            "color": "#646464",
                                                          }
                                        } />
                    </div>
          </div>
          <div
                    className="stat_y6hf3w">
                    2
                    <div
                              className="statIcon_1id1dbo"
                              onClick={[Function]}>
                              <i
                                        className="icon-share"
                                        style={
                                                  Object {
                                                            "color": "#646464",
                                                          }
                                        } />
                    </div>
          </div>
          <div
                    className="stat_y6hf3w">
                    3
                    <div
                              className="statIcon_1id1dbo"
                              onClick={[Function]}>
                              <i
                                        className="icon-comment"
                                        style={
                                                  Object {
                                                            "color": "#646464",
                                                          }
                                        } />
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ActivityStats
    comments={3}
    id={4}
    likes={1}
    shares={2} />,
}
`;
